mopidy/audio/tags.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import absolute_import, unicode_literals
+
+import collections
+import datetime
+import logging
+import numbers
+
+from mopidy import compat
+from mopidy.internal import log
+from mopidy.internal.gi import GLib, Gst
+from mopidy.models import Album, Artist, Track
+
+
+logger = logging.getLogger(__name__)
+
+
+def convert_taglist(taglist):
+    """Convert a :class:`Gst.TagList` to plain Python types.
+
+    Knows how to convert:
+
+    - Dates
+    - Buffers
+    - Numbers
+    - Strings
+    - Booleans
+
+    Unknown types will be ignored and trace logged. Tag keys are all strings
+    defined as part GStreamer under GstTagList_.
+
+    .. _GstTagList: https://developer.gnome.org/gstreamer/stable/\
+gstreamer-GstTagList.html
+
+    :param taglist: A GStreamer taglist to be converted.
+    :type taglist: :class:`Gst.TagList`
+    :rtype: dictionary of tag keys with a list of values.
```

Comment:
As soon as we get to Python 3 only (Mopidy 3.0?) I want to replace all of these `rtype` declarations with optional typing, e.g. `Dict[str, List]`. For now, I'm not sure we have a clear convention.